src/mistral_common/tokens/tokenizers/instruct.py

```diff
@@ -468,11 +468,16 @@ def _parse_json_content(self, content: str) -> Any:
         except json.JSONDecodeError:
             return content
 
+    def _parse_tool_content(self, content: str | list[TextChunk]) -> Any:
+        if isinstance(content, list):
+            content = "".join(chunk.text for chunk in content)
```

Comment:
Here it is joining with "" instead of "\n\n" in normalization.

Think both are valid cases however "" is closer to what happens to other messages' chunks + here we expect normalization to be already done.

Reply:
Should support for TextChunks really be supported already in the V2 tokenizer? I think we should maybe only add it to V3 or even V7 no? Would be nice to only add functionality to model versions that really need it.

What do you think is the "first" model that we OSS'ed that needs to encode TextChunks in the ToolMessage?